operators/src/source/ogr_source.rs

```diff
@@ -496,42 +525,64 @@ where
 
         let mut dataset_options = DatasetOptions::default();
 
+        let headers = format!(
+            "HEADERS={}",
+            columns
+                .csv_headers
+                .as_ref()
+                .map_or("AUTO", CsvHeaders::option_value)
+        );
+
         // TODO: make column x optional or allow other indication for data collection
         if columns.x.is_empty() {
+            let open_opts = &[
+                headers.as_str(),
+                // "AUTODETECT_TYPE=YES", // This breaks tests
+            ];
+            dataset_options.open_options = Some(open_opts);
             return Ok(Dataset::open_ex(&dataset_info.file_name, dataset_options)?);
         }
 
         if let Some(y) = &columns.y {
             let open_opts = &[
                 &format!("X_POSSIBLE_NAMES={}", columns.x),
                 &format!("Y_POSSIBLE_NAMES={}", y),
+                headers.as_str(),
                 "AUTODETECT_TYPE=YES",
             ];
-
             dataset_options.open_options = Some(open_opts);
             return Ok(Dataset::open_ex(&dataset_info.file_name, dataset_options)?);
         }
 
         let open_opts = &[
             &format!("GEOM_POSSIBLE_NAMES={}", columns.x),
+            headers.as_str(),
             "AUTODETECT_TYPE=YES",
         ];
         dataset_options.open_options = Some(open_opts);
-
         Ok(Dataset::open_ex(&dataset_info.file_name, dataset_options)?)
     }
 
+    // TODO: reliably detect CSV files or allow defining them as such in params
```

Comment:
I guess `CSV:` is the only way to reliably load CSV files with OGR so I guess we could remove this TODO.

Reply:
32d66d3b8da3971ea60747ad73c0c63b653e5794